service/src/main/java/com/epam/esm/service/GiftCertificateService.java

```diff
@@ -0,0 +1,180 @@
+package com.epam.esm.service;
+
+
+import com.epam.esm.convert.Convert;
+import com.epam.esm.dto.GiftAndTagDto;
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.exception.BadRequestException;
+import com.epam.esm.exception.ResourceNotFoundException;
+import com.epam.esm.repository.GiftCertificateDao;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.lang.reflect.Field;
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+@Service
+@RequiredArgsConstructor
+public class GiftCertificateService {
+
+    private final TagService tagService;
+    private final GiftsTagsService giftsTagsService;
+    private final GiftCertificateDao giftCertificateDao;
+    private final Convert<GiftCertificate, GiftAndTagDto> convert;
+
+    /**
+     * Send request for getting all Certificates with their tags
+     *
+     * @return List of Dto which contains Certificates and arrays of tag names *
+     */
+    public List<GiftAndTagDto> getCertificates() {
+        List<GiftAndTagDto> list = giftCertificateDao.findAll();
+        if (list.isEmpty()) {
```

Comment:
the code from 'if' up to 'throw new' looks boilerplate and can be moved to the separate method